src/levanter/models/diva.py

```
@@ -0,0 +1,342 @@
+        self,
+        lm_model_cls: Type[ModelWithHfSerializationMixin],
+        ref: Optional[Union[str, RepoRef]] = None,
+        config: Optional[HFCompatConfig] = None,
+        axis_mapping: Optional[ResourceMapping] = None,
+        resize_vocab_to_match_tokenizer: bool = True,
+        dtype: Optional[jnp.dtype] = None,
+    ) -> ModelWithHfSerializationMixin:
+        lev_model: DivaModel = super().load_pretrained(
+            DivaModel, ref, config, axis_mapping, resize_vocab_to_match_tokenizer, dtype
+        )  # type: ignore[assignment]
+        llm: Union[LlamaLMHeadModel | MistralLMHeadModel | GemmaLMHeadModel] = HFCheckpointConverter(
+            type(lev_model.config.dec_config),
+            lev_model.config.reference_decoder,
+        ).load_pretrained(
+            lev_model.config.dec_config.model_type,
+            lev_model.config.reference_decoder,
+            lev_model.config.dec_config,
+            axis_mapping,
+            resize_vocab_to_match_tokenizer,
+            dtype,
+        )  # type: ignore[assignment]
+        whisper: WhisperModel = HFCheckpointConverter(
+            WhisperConfig, lev_model.config.reference_encoder
+        ).load_pretrained(
+            WhisperModel,
+            lev_model.config.reference_encoder,
+            lev_model.config.enc_config,
+            axis_mapping,
+            resize_vocab_to_match_tokenizer,
+            dtype,
+        )  # type: ignore[assignment]
+        lev_model.encoder = whisper.encoder
+        lev_model.decoder = llm
+
+        return lev_model
+
+
+def load_correct_config(reference_decoder):
+    model_id = reference_decoder.lower()
+    hf_config = HfConfig.from_pretrained(reference_decoder)
+    if "llama" in model_id:
+        config = LlamaConfig.from_hf_config(hf_config)
+    elif "gemma" in model_id:
+        config = GemmaConfig.from_hf_config(hf_config)
+    elif "mistral" in model_id:
+        config = MistralConfig.from_hf_config(hf_config)
```

should you raise a better error if it's none of these?